thenewboston_node/business_logic/blockchain/base/blockchain_state.py

```diff
@@ -154,8 +154,11 @@ def generate_blockchain_state(self, last_block_number: Optional[int] = None) ->
             blockchain_state.last_block_timestamp = block.message.timestamp
             blockchain_state.next_block_identifier = block.hash
 
-        if primary_validator.identifier == get_node_identifier():
-            blockchain_state.sign(signing_key=get_node_signing_key())
+        node_signing_key = self.node_signing_key  # type: ignore
```

Comment:
This is how I think it should be implemented (it is an extension of what I previously asked you to do):
1. init parameter name is `blockchain_state_signing_key`
2. blockchain state can be created even if blockchain does not contain a primary validator (there is a code above that does not let to sign the blockchain if it does not have a primary validator)
3. blockchain state is signed by `blockchain_state_signing_key` if it is provided and not signed if it is not
4. BlochchainState class has methods `is_signed()` and `is_signed_by_primary_validator()`
5. We need to fix `def get_instance(cls: Type[T]) -> T:` to provide `blockchain_state_signing_key`